src/resolvers/utils.ts

```diff
@@ -15,3 +17,26 @@ export function getSavedItemMapFromTags(tags) {
   });
   return savedItemMap;
 }
+
+/**
+ * function to convert savedItemTagsInput list to a
+ * map of savedItemId and its unique tag names.
+ * @param input savedItemInput list
+ * @returns map with savedItemId and its unique tagNames
+ */
+export function getSavedItemTagsMap(input: SavedItemTagsInput[]): {
+  [savedItemId: string]: string[];
+} {
+  const savedItemTagsMap = input.reduce((savedItemTags, input) => {
+    let tags = input.tags;
+    if (savedItemTags[input.savedItemId]) {
+      tags = [...savedItemTags[input.savedItemId], ...input.tags];
+    }
+    return {
+      ...savedItemTags,
+      [input.savedItemId]: [...new Set([...tags])],
+    };
```

Comment:
But see my above comment -- I don't think you need to call this function at all.

Reply:
https://getpocket.atlassian.net/wiki/spaces/PE/pages/2677932103/Tagging+WIP this should not be high volume as the request needs to be limited to 30 items per mutation. thanks for catching this - we need to probably add the constraint to reject items greater than 30 items to prevent huge db read/write as we are not paginating the returns. 

we should add this max_record constraint to other bulk mutations too

Reply:
We could do this on the `executeMutation` method?

Reply:
ya can do this in a follow up PR